frontend/src/features/map_planning/components/timeline/TimelineDatePicker.tsx

```diff
@@ -0,0 +1,314 @@
+        (yearSliderItem) => newMonthItem.year === yearSliderItem.year,
+      );
+      if (newYearItem) {
+        setSelectedYearItem(newYearItem);
+      }
+    }
+
+    if (selectedDayItem.month !== newMonthItem.month) {
+      selectNewDayWhenMonthChanged(newMonthItem);
+    }
+  };
+
+  const handleYearChange = (yearItemKey: number) => {
+    const newYearItem = yearlyTimeLineEvents[yearItemKey];
+    setSelectedYearItem(newYearItem);
+
+    if (selectedMonthItem.year !== newYearItem.year) {
+      const newMonthItem = monthSliderItems.find(
+        (month) => newYearItem.year === month.year && month.month === selectedMonthItem.month,
+      );
+
+      if (newMonthItem) {
+        setVisibleMonths(calculateVisibleMonths(newYearItem));
+        setSelectedMonthItem(newMonthItem);
+
+        if (
+          selectedDayItem.month !== newMonthItem.month ||
+          selectedDayItem.year !== newMonthItem.year
+        ) {
+          selectNewDayWhenMonthChanged(newMonthItem);
+        }
+      }
+    }
+  };
+
+  const selectNewDayWhenMonthChanged = (newMonthItem: MonthItem) => {
+    const sameDayInNewMonth = daySilderItems.find(
+      (day) =>
+        newMonthItem.month === day.month &&
+        newMonthItem.year === day.year &&
+        day.day === selectedDayItem.day,
+    );
+
+    const newDay =
+      sameDayInNewMonth || getLastDayItemOfMonth(newMonthItem.month, newMonthItem.year);
+    if (newDay) {
+      setVisibleDays(calculateVisibleDays(newDay));
+      setSelectedDay(newDay);
+    }
+  };
+
+  const handleMontSliderLeftEndReached = () => {
+    if (visibleMonths[0].key > 0) {
+      setVisibleMonths(calculateVisibleMonths(selectedYearItem));
+    }
+  };
+
+  const handleMontSliderRightEndReached = () => {
+    if (visibleMonths[visibleMonths.length - 1].key < monthSliderItems.length - 1) {
+      setVisibleMonths(calculateVisibleMonths(selectedYearItem));
+    }
+  };
+
+  const handleDaySliderLeftEndReached = () => {
+    if (visibleDays[0].key > 0) {
+      setVisibleDays(calculateVisibleDays(selectedDayItem));
+    }
+  };
+
+  const handleDaySliderRightEndReached = () => {
+    if (visibleDays[visibleDays.length - 1].key < daySilderItems.length - 1) {
+      setVisibleDays(calculateVisibleDays(selectedDayItem));
+    }
+  };
+
+  function triggerDateChangedInGuidedTour(): void {
+    const changeDateEvent = new Event('dateChanged');
+    document.getElementById('timeline')?.dispatchEvent(changeDateEvent);
+  }
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      const newDay = selectedDayItem;
+      const formattedMonth = String(newDay.month).padStart(2, '0');
+      const formattedDay = String(newDay.day).padStart(2, '0');
+      const formattedDate = `${newDay.year}-${formattedMonth}-${formattedDay}`;
+      onSelectDate(formattedDate);
+      triggerDateChangedInGuidedTour();
+    }, 500);
+    return () => clearTimeout(timeoutId);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
@Bushuo can you check if this is fine?

Reply:
If onSelectDate somehow changes, the effect would not trigger.
I have to look more into it. Maybe this is a valid usecase, to not trigger the effect.
Otherwise its fine